### ISSUE TITLE
FINS PLC IOC - enable auto-reconnect

### ISSUE THREAD

**Issue by FreddieAkeroyd** [ISISComputingGroup/IBEX#2103]
The FINS plc IOC should attempt to re-connect on error, I'm guessing it doesn't do this as it was originally written to use UDP and such an explicit reconnection is not required there. If the PLC is power cycled, currently you get a lot of send() TCP failure messages

**Comment by John-Holt-Tessella** [comment]
Might have our own PLC to test this on in R78 test lab

**Comment by kjwoodsISIS** [comment]
Pair program with @FreddieAkeroyd

**Comment by John-Holt-Tessella** [comment]
May also be useful to looka t topics "Force TCP/IP reconnect from Asyn/Streamdevice" on tech talk

**Comment by FreddieAkeroyd** [comment]
Thanks, this thread is about disconnecting on read timeout and what we get is an "error" rather than a read timeout. Looking at the asyn code it should be disconnecting on error anyway.

**Comment by FreddieAkeroyd** [comment]
Reminding myself how the fins works, it does not use drvAsynIPPortConfigure

**Comment by FreddieAkeroyd** [comment]
To test you need access to a PLC - see PR

**Comment by KathrynBaker** [comment]
I have sent an email to someone appropriate to try and find out which PLC we can use for this

**Comment by KathrynBaker** [comment]
Suggested option is to the use the ZOOM vacuum PLC, and is still being commissioned. Otherwise it would be a pull the network cable to test this (if that would be a valid test)

**Comment by FreddieAkeroyd** [comment]
This could be tested on any current PLC using a simple DB file that just reads the  :MODEL:CPU   PV for example

**Comment by FreddieAkeroyd** [comment]
@GDH-ISIS please can you review this

**Comment by GDH-ISIS** [comment]
I have attempted to review this ticket using the IMAT PLC. IOC connects fine. Numbers appear valid and there appear to be no problems. HEARTBEAT increments as one would expect. 

Disconnected network cable from back of computer : 
IOC goes into failed comms state. 
HEARTBEAT keeps incrementing as one would expect so the IOC is still running.
TE:NDW1031:ATTEN:TPG300_2:B1:MEAS1EXP becomes READ INVALID.

On reconnecting the network cable, the IOC still maintains a disconnected state reporting :
2017/07/17 12:14:47.811 send_fins_header: port PLC, send() failed with An existing connection was forcibly closed by the remote host. .
2017/07/17 12:14:47.819 TE:NDW1031:ATTEN:TPG300_2:A1:MEAS2EXP devAsynInt32 process read error
2017/07/17 12:14:47.827 send_fins_header: port PLC, send() failed with An existing connection was forcibly closed by the remote host. .

Confirmed that stopping and starting the IOC re-connects - it does. Confirmed that the computer IP address is re-established to the same IP address using ipconfig. Confirmed that the modifications to finsUDP.c file are used. Also did make clean uninstall on support/FINS and on ioc/master/FINS. Re-built both locations and confirmed that the problem persisted.
I do not believe this is re-connecting as one would expect.
Note : I am not sure the errors on the PVs are propagating through properly for the IMAT PLC.
